packages/mds-api-server/middleware/prometheus/provider-id-labeler.ts

```diff
@@ -0,0 +1,19 @@
+import type { Labels as PrometheusLabels } from 'express-prom-bundle'
+import { ApiRequest, ApiResponse, ApiResponseLocals, ApiClaims } from '../../@types'
+
+const providerIdLabelTransformer = <AccessTokenScope extends string>(
+  labels: PrometheusLabels,
+  req: ApiRequest,
+  res: ApiResponse & ApiResponseLocals<ApiClaims<AccessTokenScope>>
+) => {
+  if (res.locals.claims?.provider_id) {
+    return Object.assign(labels, { provider_id: res.locals.claims.provider_id })
+  }
+  return labels
+}
+
+export const ProviderIdLabeler = {
+  label: 'provider_id',
+  base: null,
+  transformer: providerIdLabelTransformer
+}
```

Comment:
Should this be typed?
```suggestion
export const ProviderIdLabeler: PrometheusLabeler = {
  label: 'provider_id',
  base: null,
  transformer: providerIdLabelTransformer
}
```

Reply:
It certainly could be, I was just relying on the typechecking when actually _using_ it. Can add the manual annotation, though.